Foundation/NSURLProtocol.swift

```diff
@@ -142,6 +145,92 @@ public protocol URLProtocolClient : NSObjectProtocol {
     func urlProtocol(_ protocol: URLProtocol, didCancel challenge: URLAuthenticationChallenge)
 }
 
+internal class _ProtocolClient : NSObject, URLProtocolClient {
+
+    func urlProtocol(_ protocol: URLProtocol, didReceive response: URLResponse, cacheStoragePolicy policy: URLCache.StoragePolicy) {
+        `protocol`.task?.setResponse = response
+    }
+
+    func urlProtocolDidFinishLoading(_ protocol: URLProtocol) {
+        guard let session = `protocol`.task?.session as? URLSession else { fatalError() }
+        switch session.behaviour(for: `protocol`.task!) {
```

Comment:
May be we should set a guard  for `protocol.task` and generate a `fatalError` if its is nil?  We can then access task in the remainder of the method without worries of nil :)